File: client/iih_stub/iih.ci_stub/iih.ci.ord/ciprn/d/CiPrnItmDO.cs

using System;
using xap.mw.core.data;
using xap.mw.coreitf.d;

namespace iih.ci.ord.ciprn.d
{
    /// <summary>
    /// CiPrnItmDO 的摘要说明。
    /// </summary>
    public class CiPrnItmDO : BaseDO {

		public const string TABLE_NAME = "ci_prn_item";
		public const string TABLE_ALIAS_NAME = "a1";

        public CiPrnItmDO() {
        }
		public string Id_ciprnitm {
            get { return getAttrVal<string>("Id_ciprnitm",null); }
            set { setAttrVal<string>("Id_ciprnitm", value); }
        }
		public string Id_ciprn {
            get { return getAttrVal<string>("Id_ciprn",null); }
            set { setAttrVal<string>("Id_ciprn", value); }
        }
		public string Id_biz {
            get { return getAttrVal<string>("Id_biz",null); }
            set { setAttrVal<string>("Id_biz", value); }
        }
		public FDouble Amt_biz {
            get { return getAttrVal<FDouble>("Amt_biz",null); }
            set { setAttrVal<FDouble>("Amt_biz", value); }
        }
        public int Ds {
            get { return getAttrVal<int>("Ds",0);}
            set { setAttrVal<int>("Ds", value); }
        }

        public DateTime? Sv        {
            get { return getAttrVal<FDateTime>("Sv",null); }
            set { setAttrVal<FDateTime>("Sv", value); }
        }
        
        /// <summary>
        /// 返回表名
        /// </summary>
        /// <returns></returns>
        public override string getTableName()
        {
            return "ci_prn_item";
        }
        
        /// <summary>
        /// 返回表别名
        /// </summary>
        /// <returns></returns>
        public override string getAliasTableName()
        {
            return "a1";
        }
        
        /// <summary>
        /// 返回主键字段名
        /// </summary>
        /// <returns></returns>
        public override string getPKFieldName()
        {
            return "Id_ciprnitm";
        }

        /// <summary>
        /// 返回全路径DO类名
        /// </summary>
        /// <returns></returns>
        public override string getFullClassName()
        {
            return "iih.ci.ord.ciprn.d.CiPrnItmDO";
        }

        /// <summary>
        /// 返回属性列表
        /// </summary>
        /// <returns></returns>
        public override string[] getAttrNames()
        {
            return new string[]{"Ds","Sv",  "Id_ciprnitm", "Id_ciprn", "Id_biz", "Amt_biz"};
        }
        
    }
}

File: server/iih_pub/iih.ci_pub/src/main/java/iih/ci/ord/ciprn/d/CiPrnItmDO.java
package iih.ci.ord.ciprn.d;

import xap.sys.appfw.orm.desc.DescManager;
import xap.mw.coreitf.c.GlobalConst;
import xap.mw.coreitf.i.IDODesc;
import xap.mw.coreitf.d.*;
import xap.mw.core.data.*;
import iih.ci.ord.ciprn.d.desc.CiPrnItmDODesc;
import java.math.BigDecimal;

/**
 * 临床打印单明细 DO数据 
 * 
 */
public class CiPrnItmDO extends BaseDO {
	private static final long serialVersionUID = 1L;

	public static final String ID_CIPRNITM= "Id_ciprnitm";
	public static final String ID_CIPRN= "Id_ciprn";
	public static final String ID_BIZ= "Id_biz";
	public static final String AMT_BIZ= "Amt_biz";
	public static final String DS = "Ds";
	public static final String SV = "Sv";
	
	public String getId_ciprnitm() {
		return ((String) getAttrVal("Id_ciprnitm"));
	}	
	public void setId_ciprnitm(String Id_ciprnitm) {
		setAttrVal("Id_ciprnitm", Id_ciprnitm);
	}
	public String getId_ciprn() {
		return ((String) getAttrVal("Id_ciprn"));
	}	
	public void setId_ciprn(String Id_ciprn) {
		setAttrVal("Id_ciprn", Id_ciprn);
	}
	public String getId_biz() {
		return ((String) getAttrVal("Id_biz"));
	}	
	public void setId_biz(String Id_biz) {
		setAttrVal("Id_biz", Id_biz);
	}
	public FDouble getAmt_biz() {
		return ((FDouble) getAttrVal("Amt_biz"));
	}	
	public void setAmt_biz(FDouble Amt_biz) {
		setAttrVal("Amt_biz", Amt_biz);
	}

	public Integer getDs() {
		return ((Integer) getAttrVal(GlobalConst.DELETE_SIGN));
	}
	
	public void setDs(Integer ds) {
		setAttrVal(GlobalConst.DELETE_SIGN, ds);
	}
	
	public FDateTime getSv() {
		return ((FDateTime) getAttrVal(GlobalConst.SYSTEM_VERSION));
	}
	
	public void setSv(FDateTime sv) {
		setAttrVal(GlobalConst.SYSTEM_VERSION, sv);
	}

//	@Override
//	public java.lang.String getParentPKFieldName() {
//		return null;
//	}
  
	@Override
	public String getPKFieldName() {
		return "Id_ciprnitm";
	}
	
	@Override
	public String getTableName() {	  
		return "ci_prn_item";
	}
	
	@Override
	public IDODesc getDODesc() {
		return DescManager.getInstance().getDODesc(CiPrnItmDODesc.class);
	}
	
}